bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -292,12 +308,28 @@ public EntryLogger(ServerConfiguration conf,
                 logId = lastLogId;
             }
         }
-        this.leastUnflushedLogId = logId + 1;
+        this.recentlyCreatedEntryLogsStatus = new RecentEntryLogsStatus(logId + 1);
         this.entryLoggerAllocator = new EntryLoggerAllocator(logId);
-        this.conf = conf;
-        this.entryLogPerLedgerEnabled = conf.isEntryLogPerLedgerEnabled();
-
-        initialize();
+        if (entryLogPerLedgerEnabled) {
+            this.entryLogManager = new EntryLogManagerForSingleEntryLog() {
+                @Override
+                public void checkpoint() throws IOException {
+                    /*
+                     * In the case of entryLogPerLedgerEnabled we need to flush
+                     * both rotatedlogs and currentlogs. This is needed because
+                     * syncThread periodically does checkpoint and at this time
+                     * all the logs should be flushed.
+                     *
+                     */
+                    if (entryLogPerLedgerEnabled) {
```

Comment:
nit: it should be simply, (will fix it in next iteration)

public void checkpoint() throws IOException {
    flushCurrentLogs();
    super.checkpoint();
}